File: WoofIOS/SwiftPackages/Domain/UseCase/Sources/UseCase/Moderation/ReportDogUseCase.swift
import Combine
import DI
import Models
import Repositories

@Factory
public final class ReportDogUseCase {
    private let moderationRepository: ModerationRepository

    public func callAsFunction(dog: Dog) -> AnyPublisher<Void, ModerationError> {
        moderationRepository.addReport(dog: dog)
    }
}

File: WoofAndroid/domain/usecase/src/main/kotlin/com/perrystreet/woof/usecase/moderation/ReportDogUseCase.kt
package com.perrystreet.woof.usecase.moderation

import com.perrystreet.woof.models.dog.Dog
import com.perrystreet.woof.repositories.moderation.ModerationRepository
import io.reactivex.rxjava3.core.Completable
import org.koin.core.annotation.Factory

@Factory
class ReportDogUseCase(
    private val moderationRepository: ModerationRepository,
) {
    operator fun invoke(dog: Dog): Completable = moderationRepository.addReport(dog)
}
